src/drive/web/modules/trash/Toolbar.spec.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { render, fireEvent, screen, waitFor } from '@testing-library/react'
+import flag from 'cozy-flags'
+import { createMockClient } from 'cozy-client'
+
+import AppLike from 'test/components/AppLike'
+import { Toolbar } from './Toolbar'
+import EmptyTrashConfirm from './components/EmptyTrashConfirm'
+
+describe('Toolbar', () => {
+  const client = createMockClient({})
+  client.collection = jest.fn(() => client)
+  client.emptyTrash = jest.fn()
+
+  const mockModalContextValue = {
+    pushModal: jest.fn(),
+    popModal: jest.fn()
+  }
+
+  beforeEach(() => {
+    flag('drive.client-migration.enabled', true)
+  })
+
+  afterEach(() => {
+    flag('drive.client-migration.enabled', null)
+  })
+
+  it('asks for confirmation before emptying the trash', () => {
+    const { getByText } = render(
+      <AppLike client={client} modalContextValue={mockModalContextValue}>
+        <Toolbar
+          t={jest.fn(x => x)}
+          disabled={false}
+          selectionModeActive={false}
+          emptyTrash={jest.fn()}
+          breakpoints={{ isMobile: false }}
+        />
+      </AppLike>
+    )
+
+    const emptyTrashButton = getByText('toolbar.empty_trash')
+    fireEvent.click(emptyTrashButton)
+
+    const actuallyCalledModal = mockModalContextValue.pushModal.mock.calls[0][0]
+    expect(mockModalContextValue.pushModal).toHaveBeenCalledWith(
+      <EmptyTrashConfirm
+        onConfirm={actuallyCalledModal.props.onConfirm}
+        onClose={actuallyCalledModal.props.onClose}
+      />
+    )
+
+    actuallyCalledModal.props.onConfirm()
```

Comment:
Why not `fireEvent.click(getByText(confirm))` (i don't know the exact string)and then check if emptyTrash has been called?

Reply:
Because we have mocked the modal context and aren't rendering the ModalStack, so the confirmation modal is never actually rendered during the test.

Reply:
So... Why are we mocking the Modal context? :D Why not been as close as possible than the use of this component ? And instead of checking if the pushModal was called with the right component, we check if we have the text "are you sure to delete the files" ?